community/common/src/main/java/org/neo4j/memory/GlobalMemoryTracker.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.memory;
+
+import java.util.concurrent.atomic.LongAdder;
+
+/**
+ * Global memory tracker that can be used in a global multi threaded context to record
+ * allocation and de-allocation of native memory.
+ * @see org.neo4j.memory.MemoryAllocationTracker
+ * @see MemoryTracker
+ */
+public class GlobalMemoryTracker implements MemoryTracker, MemoryAllocationTracker
+{
+    private LongAdder longAdder = new LongAdder();
+
+    public static final GlobalMemoryTracker INSTANCE = new GlobalMemoryTracker();
+
+    private GlobalMemoryTracker()
+    {
```

Comment:
Follow the best practice and `throw new AssertionError`.

Reply:
thats not really possible since we still need one instance :)

Reply:
Silly me :see_no_evil: